pexec/managed_process.go

```diff
@@ -187,7 +197,24 @@ func (p *managedProcess) manage(stdOut, stdErr io.ReadCloser) {
 					}
 					return
 				}
-				p.logger.Debugw("output", "name", name, "data", string(line))
+				if p.shouldLog {
+					p.logger.Debugw("output", "name", name, "data", string(line))
+				}
+				if p.logWriter != nil && !logWriterError {
+					_, err := (*p.logWriter).Write(line)
+					if err == nil {
+						_, err = (*p.logWriter).Write([]byte("\n"))
```

Comment:
add continue here so you can remove err != nil and dedent block

Reply:
I don't think I should add continue here, since I want to check if either write had an error.

Reply:
oh I see.

so a io.Writer writes should be a DEBUG and are very exceptional